packager/packager.js

```diff
@@ -127,23 +127,30 @@ function openStackFrameInEditor(req, res, next) {
 }
 
 function getDevToolsLauncher(options) {
+  var connected = false;
   return function(req, res, next) {
     if (req.url === '/debugger-ui') {
       var debuggerPath = path.join(__dirname, 'debugger.html');
       res.writeHead(200, {'Content-Type': 'text/html'});
       fs.createReadStream(debuggerPath).pipe(res);
-    } else if (req.url === '/launch-chrome-devtools') {
-      var debuggerURL = 'http://localhost:' + options.port + '/debugger-ui';
-      var script = 'launchChromeDevTools.applescript';
-      console.log('Launching Dev Tools...');
-      exec(path.join(__dirname, script) + ' ' + debuggerURL, function(err, stdout, stderr) {
-        if (err) {
-          console.log('Failed to run ' + script, err);
-        }
-        console.log(stdout);
-        console.warn(stderr);
-      });
-      res.end('OK');
+    } else if (req.url === '/launch-browser-devtools') {
+      if (connected) {
+        console.log('Already connect to Dev Tools in default browser...');
+        res.end('OK');
+      }
+      else {
+        var debuggerURL = 'http://localhost:' + options.port + '/debugger-ui';
+        console.log('Launching Dev Tools in default browser...');
+        connected = true;
```

Comment:
So the two use cases I’m currently missing are:

1. Cmd+Ctrl+Z switches back to running js on the device. This should reset the connected state so future Cmd+D can reconnect to a browser tab.
2. Restarting the simulator from XCode without closing the window should be able to reconnect.

Am I missing anything else?